pkg/tfvars/gcp/gcp.go

```diff
@@ -137,3 +169,121 @@ func generateDiskEncryptionKeyLink(keyRef *machineapi.GCPEncryptionKeyReference,
 
 	return fmt.Sprintf(kmsKeyNameFmt, projectID, keyRef.KMSKey.Location, keyRef.KMSKey.KeyRing, keyRef.KMSKey.Name)
 }
+
+// getTagValuesNames returns list of tag values Names as required by terraform APIs
+// fetched using the user provided tag ParentID, Key shortname and Value shortname.
+func getTagValuesNames(cred, projID string, tags []gcp.UserTag) (map[string]string, error) {
```

Comment:
I don't think this logic belongs in tfvars. tfvars is primarily concerned with converting inputs into the proper terraform variable formats. One typical pattern, which applies here, is to read from the machine manifests.

Reply:
Terraform tag APIs accept tag value in the form `tagValues/<tag_value_id>`, which can be obtained using the user provided input `ParentID`, `Tag_Key_Name`, `Tag_Value_Name` and this is required just for the terraform APIs, golang APIs accept the above form as well as the `<ParentID>/<Tag_Key_Name>/<Tag_Value_Name>`. so I thought to keep it in this source file.

Please suggest.

Reply:
suggestion incorporated.